src/overlays/actors/ovl_Bg_Ice_Shutter/z_bg_ice_shutter.c

```diff
@@ -27,19 +31,104 @@ const ActorInit Bg_Ice_Shutter_InitVars = {
+        (Math_Coss(thisx->dyna.actor.shape.rot.y) * sp24) + thisx->dyna.actor.initPosRot.pos.z;
+}
+
+void BgIceShutter_Init(Actor* thisx, GlobalContext* globalCtx) {
+    f32 pad;
+    f32 sp24;
+    s32 localC;
+    s32 sp28;
+    f32 temp_f6;
+
+    localC = 0;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyInfo_SetActorMove(&THIS->dyna, 0);
+    sp28 = thisx->params & 0xFF;
+    thisx->params = (thisx->params >> 8) & 0xFF;
+    DynaPolyInfo_Alloc(&D_06002854, &localC);
+    THIS->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, localC);
+    if (sp28 == 2) {
+        thisx->shape.rot.x = (s16)(-0x4000);
+    }
+
+    if (sp28 != 1) {
+        if (Flags_GetClear(globalCtx, thisx->room) != 0) {
+            Actor_Kill(thisx);
+        } else {
+            THIS->actionFunc = &func_80891CF4;
+        }
+
+    } else if (Flags_GetSwitch(globalCtx, thisx->params) != 0) {
+        Actor_Kill(thisx);
+    } else {
+        THIS->actionFunc = &func_80891D6C;
+    }
+
+    if (sp28 == 2) {
+        temp_f6 = Math_Sins(thisx->shape.rot.x) * 50.0f;
+        thisx->posRot2.pos.x = (Math_Sins(thisx->shape.rot.y) * temp_f6) + thisx->initPosRot.pos.x;
+        thisx->posRot2.pos.y = thisx->initPosRot.pos.y;
+        thisx->posRot2.pos.z = thisx->initPosRot.pos.z + (Math_Coss(thisx->shape.rot.y) * temp_f6);
+        return;
+    }
+
+    Actor_SetHeight(thisx, 50.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/BgIceShutter_Destroy.s")
+void BgIceShutter_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, THIS->dyna.dynaPolyId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/func_80891CF4.s")
+void func_80891CF4(BgIceShutter* thisx, GlobalContext* globalCtx) {
+    if (Flags_GetTempClear(globalCtx, thisx->dyna.actor.room) != 0) {
+        Flags_SetClear(globalCtx, thisx->dyna.actor.room);
+        Audio_PlaySoundAtPosition(globalCtx, &(thisx->dyna.actor.posRot), 0x1E, 0x2814);
+        thisx->actionFunc = &func_80891DD4;
+        if (thisx->dyna.actor.shape.rot.x == 0) {
+            func_80080480(globalCtx, thisx);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/func_80891D6C.s")
+void func_80891D6C(BgIceShutter* thisx, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, thisx->dyna.actor.params) != 0) {
+        Audio_PlaySoundAtPosition(globalCtx, &(thisx->dyna.actor.posRot), 0x1E, 0x2814);
+        thisx->actionFunc = &func_80891DD4;
+        func_80080480(globalCtx, thisx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Shutter/func_80891DD4.s")
+void func_80891DD4(BgIceShutter* thisx, GlobalContext* globalCtx) {
+    Math_ApproxF(&thisx->dyna.actor.speedXZ, 30.0f, 2.0f);
+    if (Math_ApproxF(&thisx->dyna.actor.velocity.y, 210.0f, thisx->dyna.actor.speedXZ) != 0) {
```

Comment:
`Math_ApproxF` also returns a boolean (1 if the approximation is "close enough")